### PR TITLE
LPS-165008 | LPS-165011 | LPS-165012

### PR DESCRIPTION
**Pull request by murilostodolni**

### CODE REVIEW

**Comment by murilostodolni** [comment]
Hi @carolmariaabb can you review from that [commit](https://github.com/liferay-objects/liferay-portal/pull/1289/commits/998e38fdb34e5f3e204ac362d83783d2b1c9eacb) down, please?

**Comment by murilostodolni** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 1]
```diff
@@ -1021,6 +1021,8 @@ public void testUpdateCustomObjectField() throws Exception {
 		_testUpdateCustomObjectField(
 			objectFieldBuilder.businessType(
 				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST
+			).listTypeDefinitionId(
+				_listTypeDefinition.getListTypeDefinitionId()
 			).dbType(
 				ObjectFieldConstants.DB_TYPE_STRING
 			).indexedAsKeyword(
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1028]: @murilostodolni is it possible to sort this alphabetically?

File: modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/service/test/ListTypeDefinitionLocalServiceTest.java [hunk 2]
```diff
@@ -57,13 +58,11 @@ public class ListTypeDefinitionLocalServiceTest {
...
-		ListTypeDefinition listTypeDefinition =
-			_listTypeDefinitionLocalService.addListTypeDefinition(
-				TestPropsValues.getUserId(),
-				Collections.singletonMap(
-					LocaleUtil.US, RandomTestUtil.randomString()));
+		ListTypeDefinition listTypeDefinition = _addListTypeDefinition();
 
 		Assert.assertNotNull(listTypeDefinition);
+		Assert.assertNotNull(listTypeDefinition.getExternalReferenceCode());
+		Assert.assertNotNull(listTypeDefinition.getName());
```
Comment (carolmariaabb) [inline modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/service/test/ListTypeDefinitionLocalServiceTest.java:85]: @murilostodolni `assertNotNull` only checks if it is not `null`? Wouldn't it be better to check that it's not empty?

```
Assert.assertTrue(Validator.isNotNull(listTypeDefinition.getExternalReferenceCode()));
Assert.assertTrue(Validator.isNotNull(llistTypeDefinition.getName()));
```

File: modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/service/test/ListTypeDefinitionLocalServiceTest.java [hunk 3]
```diff
@@ -144,6 +143,42 @@ public void testDeleteListTypeDefinition() throws Exception {
...
+			name, listTypeDefinition.getName(LocaleUtil.getDefault()));
+
+		listTypeDefinition =
+			_listTypeDefinitionLocalService.updateListTypeDefinition(
+				StringPool.BLANK, listTypeDefinition.getListTypeDefinitionId(),
+				Collections.singletonMap(LocaleUtil.getDefault(), name));
+
+		Assert.assertNotEquals(
+			listTypeDefinitionERC,
+			listTypeDefinition.getExternalReferenceCode());
```
Comment (carolmariaabb) [inline modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/service/test/ListTypeDefinitionLocalServiceTest.java:192]: @murilostodolni can we check that `listTypeDefinition.getExternalReferenceCode()` is not empty?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
Great job, @murilostodolni!

Just a few comments.

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
Failures seem to be related.

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
Manually forwarded https://github.com/brianchandotcom/liferay-portal/pull/125629

**Comment by carolmariaabb** [comment]
ci:close

**Pull request closed without merging**